### PR TITLE
Updates the notification policy documentation by adding the new "update PR comments in place" feature.

### PR DESCRIPTION
**Pull request by silviutanasa**
# Description of the change

Updates the notification policy documentation by adding the new "update PR comments in place" feature.

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [x] I have reviewed the [guidelines for contributing](https://github.com/spacelift-io/user-documentation/blob/main/CONTRIBUTING.md) to this repository.
- [x] The preview looks fine.
- [x] The tests pass.
- [x] The commit history is clean and meaningful.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/user-documentation/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).
- You have updated the navigation files correctly:
    - [x] No new pages have been added, or;
    - [ ] Only _nav.yaml_ has been updated because the changes only apply to SaaS, or;
    - [ ] Only _nav.self-hosted.yaml_ has been updated because the changes only apply to Self-Hosted, or;
    - [ ] Both _nav.yaml_ and _nav.self-hosted.yaml_ have been updated.

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

File: docs/concepts/policy/notification-policy.md [hunk 1]
```diff
@@ -478,6 +479,38 @@ pull_request contains {"id": run.commit.pull_request_id} if {
 
 [View the example in the rego playground](https://play.openpolicyagent.org/p/i4hswca0Fr){: rel="nofollow"}.
 
+#### Updating a pull request comment, in place
+
+!!! info
+    This feature is currently available only for GitHub and GitHub Enterprise VCS integrations.
+    We intend to add support, for the rest of the supported VCS integrations, in the near future.
```
Comment (cube2222) [inline docs/concepts/policy/notification-policy.md:486]: Nit: both commas in this line I believe are unnecessary

File: docs/concepts/policy/notification-policy.md [hunk 2]
```diff
@@ -478,6 +479,38 @@ pull_request contains {"id": run.commit.pull_request_id} if {
...
+
+For example here is a rule which will add a comment to the pull request that triggered the run,
```
Comment (cube2222) [inline docs/concepts/policy/notification-policy.md:488]: I think we can leave the "For example " off here.

File: docs/concepts/policy/notification-policy.md [hunk 3]
```diff
@@ -478,6 +479,38 @@ pull_request contains {"id": run.commit.pull_request_id} if {
...
+and will keep updating that comment for every run state change, instead of creating new comments. This helps keeping a clean PR.<br>
+The key, for comment updates in place, is the `deduplication_key` which must have a constant value across the PR lifetime,
```
Comment (cube2222) [inline docs/concepts/policy/notification-policy.md:490]: for updating comments in place

File: docs/concepts/policy/notification-policy.md [hunk 4]
```diff
@@ -478,6 +479,38 @@ pull_request contains {"id": run.commit.pull_request_id} if {
...
 [View the example in the rego playground](https://play.openpolicyagent.org/p/i4hswca0Fr){: rel="nofollow"}.
 
+#### Updating a pull request comment, in place
+
+!!! info
+    This feature is currently available only for GitHub and GitHub Enterprise VCS integrations.
+    We intend to add support, for the rest of the supported VCS integrations, in the near future.
+
+For example here is a rule which will add a comment to the pull request that triggered the run,
+and will keep updating that comment for every run state change, instead of creating new comments. This helps keeping a clean PR.<br>
```
Comment (cube2222) [inline docs/concepts/policy/notification-policy.md:489]: > This helps keeping a clean PR.

I'd remove that part.

File: docs/concepts/policy/notification-policy.md [hunk 5]
```diff
@@ -478,6 +479,38 @@ pull_request contains {"id": run.commit.pull_request_id} if {
 
 [View the example in the rego playground](https://play.openpolicyagent.org/p/i4hswca0Fr){: rel="nofollow"}.
 
+#### Updating a pull request comment, in place
+
+!!! info
+    This feature is currently available only for GitHub and GitHub Enterprise VCS integrations.
+    We intend to add support for the rest of the supported VCS integrations in the near future.
```
Comment (Yantrio) [inline docs/concepts/policy/notification-policy.md:486]:
```suggestion
    This feature is currently only available for GitHub and GitHub Enterprise VCS integrations.
    We intend to add support for the rest of the supported VCS integrations in the near future.
```

**Review by Yantrio — APPROVED** [review]
one tiny nitpick. otherwise it's good

**Review by Yantrio — APPROVED** [review]

**Pull request merged**